cluster_management/src/main/java/com/pinterest/rocksplicator/LeaderFollowerStateModelFactory.java

```diff
@@ -471,6 +478,46 @@ public void onBecomeFollowerFromOffline(Message message, NotificationContext con
           Utils.restoreLocalDBFromS3(adminPort, dbName, s3Bucket, s3Path, snapshotHost,
               snapshotPort);
         }
+
+        try {
+          LOG.error(dbName + " Catching up live updates to leader");
+          Map<String, String> liveHostAndRole = getLiveHostAndRole(context, dbName);
+          // Find upstream, prefer Leader
+          String upstream = null;
+          for (Map.Entry<String, String> instanceNameAndRole : liveHostAndRole.entrySet()) {
+            String role = instanceNameAndRole.getValue();
+            if (role.equalsIgnoreCase("LEADER")) {
+              upstream = instanceNameAndRole.getKey();
+              break;
+            } else {
+              upstream = instanceNameAndRole.getKey();
+            }
+          }
+
+          String upstreamHost = (upstream == null ? LOCAL_HOST_IP : upstream.split("_")[0]);
+          LOG.error(dbName + " got leader " + upstreamHost);
+          int upstreamPort =
+              (upstream == null ? adminPort : Integer.parseInt(upstream.split("_")[1]));
+
+          long local_seq_num;
+          long leader_seq_num;
+          // wait for up to 10 mins
+          for (int i = 0; i < 600; ++i) {
+            local_seq_num = Utils.getLocalLatestSequenceNumber(dbName, adminPort);
+            leader_seq_num = Utils.getLatestSequenceNumber(dbName, upstreamHost, upstreamPort);
+
+            // If leader sequence number is within threshold, consider caught up.
+            if (leader_seq_num < local_seq_num + LEADER_CATCH_UP_THRESHOLD) {
+              LOG.error(dbName + " caught up!");
+              break;
+            }
+            TimeUnit.SECONDS.sleep(1);
+          }
+        } catch (Exception e) {
```

Comment:
there is no explicit Exception throw from the catch up step, right? Then, this catch won't catch anything.

Reply:
I thought there could be exceptions while getting latest seq number from leader. But that function is already catching Thrift exceptions. I'll leave it here to catch any other exceptions, we want this catch up to be a best effort.